### PR TITLE
Fix remote configuration assertion for is_agent and config_states

### PR DESCRIPTION
**Pull request by GustavoCaso**
## Description

Fix remote configuration assertion for `is_agent` and `config_states`

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

File: tests/remote_config/test_remote_configuration.py [hunk 1]
```diff
@@ -134,15 +134,16 @@ def rc_check_request(data, expected, caching):
         ), f"targetsVersion was expected to be {expected_targets_version}, not {targets_version}"
 
         # verify that the tracer is properly storing and reporting on its config state
-        expected_config_states = client_state.get("config_states")
+        expected_config_states = expected.get("config_states")
```
Comment (GustavoCaso) [inline tests/remote_config/test_remote_configuration.py:137]: We were not using the `expected` here so we always compared the same client state

File: tests/remote_config/test_remote_configuration.py [hunk 2]
```diff
@@ -134,15 +134,16 @@ def rc_check_request(data, expected, caching):
...
         config_states = client_state.get("config_states")
-        if expected_config_states is None and config_states is not None:
+
+        if expected_config_states is None and (config_states is not None and len(config_states) > 0):
```
Comment (GustavoCaso) [inline tests/remote_config/test_remote_configuration.py:140]: Base on the RFC `config_states` can be null or empty list

File: tests/remote_config/test_remote_configuration.py [hunk 3]
```diff
@@ -110,7 +110,7 @@ def validator(data):
             client = data["request"]["content"]["client"]
             client_tracer = client["client_tracer"]
 
-            assert "is_agent" not in client, "'client.is_agent' MUST either NOT be set or set to false"
+            assert "is_agent" not in client or client['is_agent'] == False, "'client.is_agent' MUST either NOT be set or set to false"
```
Comment (GustavoCaso) [inline tests/remote_config/test_remote_configuration.py:113]: `is_agent` could be false

**Review by ameske — APPROVED** [review]

**Review by marcotc — APPROVED** [review]

**Comment by cbeauchesne** [comment]
Ok, once the test is fixed, turns out java and go seems no to be compliant. Just wait for confirmation before merging.

**Comment by cbeauchesne** [comment]
Adding @Hellzy  and @smola as we skip some tests for java and golang

File: tests/remote_config/test_remote_configuration.py [hunk 4]
```diff
@@ -249,6 +290,7 @@ class Test_RemoteConfigurationUpdateSequenceLiveDebugging(RemoteConfigurationFie
     # that spawns multiple worker processes, each running its own RCM client.
     request_number = defaultdict(int)
 
+    @bug(library="java", reason="id reported for config state is not the expected one")
```
Comment (smola) [inline tests/remote_config/test_remote_configuration.py:293]:
```suggestion
    @bug(context.library < "java@1.13.0", reason="id reported for config state is not the expected one")
```

**Review by smola — APPROVED** [review]
Fix for Java coming up on: https://github.com/DataDog/dd-trace-java/pull/5114

File: tests/remote_config/test_remote_configuration.py [hunk 5]
```diff
@@ -210,6 +248,8 @@ class Test_RemoteConfigurationUpdateSequenceFeatures(RemoteConfigurationFieldsBa
         context.library >= "java@1.4.0" and context.agent_version < "1.8.0" and context.appsec_rules_file is not None,
         reason="ASM_FEATURES was not subscribed when a custom rules file was present",
     )
+    @bug(library="golang", reason="missing update file datadog/2/ASM_FEATURES/ASM_FEATURES-third/config")
+    @bug(library="java", reason="id reported for config state is not the expected one")
```
[the remark this answers is not in the document]
Comment (smola) [reply tests/remote_config/test_remote_configuration.py:252]: This depends on https://github.com/DataDog/dd-trace-java/pull/5114 which I expected to merge today.

[the remark this answers is not in the document]
Comment (cbeauchesne) [reply tests/remote_config/test_remote_configuration.py:252]: ok, I wait for it !

[the remark this answers is not in the document]
Comment (smola) [reply tests/remote_config/test_remote_configuration.py:252]: Done.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**